Receipt/index.ts

```diff
@@ -71,80 +71,88 @@ export namespace Receipt {
-
-		const indexPages = Array.from({ length: Math.ceil(receipts.length / receiptsPerIndexPage) }).map((_, pageNumber) =>
-			receipts.slice(pageNumber * receiptsPerIndexPage, (pageNumber + 1) * receiptsPerIndexPage)
+		const indexPages = Array.from({
+			length: Math.ceil(
+				receiptData.reduce((total, costCenter) => total + costCenter.receipts.length, 0) / receiptsPerIndexPage
+			),
+		}).map((_, pageNumber) =>
+			receiptData.slice(pageNumber * receiptsPerIndexPage, (pageNumber + 1) * receiptsPerIndexPage)
 		)
-		let totalVat = 0
-		let totalNet = 0
+
 		for (const [i, indexPage] of indexPages.entries()) {
-			const page = pdfDoc.insertPage(1 + i, [width, height])
-			headers.forEach((header, index) =>
-				page.drawText(header, {
-					x: xMargin + index * cellWidth,
-					y: height - yMargin,
-					size: headerSize,
-				})
-			)
-			page.moveTo(xMargin, height - yMargin - fontSize / 2)
-			page.drawLine({
-				start: { x: xMargin, y: height - yMargin - lineThickness - lineMargin },
-				end: { x: width - xMargin, y: height - yMargin - lineThickness - lineMargin },
-				thickness: lineThickness,
-			})
-			for (const receipt of indexPage) {
-				const currency = receipt.details.total[0].net[1]
-				const [net, vat] = receipt.details.total.reduce<[number, number]>(
-					([n, v], { net: [net], vat: [vat] }) => [n + net, v + vat],
-					[0, 0]
-				)
-				const cellText = [
-					`${pdfDoc.getPageCount() + indexPages.length - i}`,
-					`${vat}`,
-					`${net}`,
-					`${net + vat}`,
-					currency,
-				]
-				page.moveDown(lineHeight)
-				cellText.forEach((text, index) => {
-					page.drawText(text, {
+			for (const CostCenter of indexPage) {
+				const page = pdfDoc.addPage([width, height])
+				page.drawText(`Summary for cost center: ${CostCenter.costCenter}`, { x: xMargin, y: height - yMargin })
+				ccStartPage[CostCenter.costCenter] = pdfDoc.getPageCount()
+				height -= 20
+				headers.forEach((header, index) =>
+					page.drawText(header, {
 						x: xMargin + index * cellWidth,
-						size: fontSize,
-						font: font,
+						y: height - yMargin,
+						size: headerSize,
 					})
+				)
+				page.moveTo(xMargin, height - yMargin - fontSize / 2)
+				page.drawLine({
+					start: { x: xMargin, y: height - yMargin - lineThickness - lineMargin },
+					end: { x: width - xMargin, y: height - yMargin - lineThickness - lineMargin },
+					thickness: lineThickness,
 				})
-				const newFile = await PDFLib.PDFDocument.load(receipt.file)
-				const copiedPages = await pdfDoc.copyPages(newFile, newFile.getPageIndices())
-				copiedPages.forEach(page => pdfDoc.addPage(page))
 
-				totalVat = isoly.Currency.add(currency, totalVat, vat)
-				totalNet = isoly.Currency.add(currency, totalNet, net)
+				for (const receipt of CostCenter.receipts) {
+					const currency = receipt.details.total[0].net[1]
+					const [net, vat] = receipt.details.total.reduce<[number, number]>(
+						([n, v], { net: [net], vat: [vat] }) => [n + net, v + vat],
```

Comment:
previously `isoly.Currency.add` was used and now it is not. This is something we all need to be better at using. will just point it out here since it was use in the code that was replaced but this should be added in MANY places (not just this compile function). We found out about it very recently so we should have it as a mini epic to make sure it is used everywhere we are combining currencies.